iceoryx_hoofs/include/iceoryx_hoofs/memory/relative_pointer.inl

```diff
@@ -189,36 +178,36 @@ template <typename T>
 // NOLINTJUSTIFICATION NewType size is comparable to an integer, hence pass by value is preferred
 // NOLINTNEXTLINE(performance-unnecessary-value-param)
 inline typename RelativePointer<T>::offset_t RelativePointer<T>::getOffset(const segment_id_t id,
-                                                                           const_ptr_t ptr) noexcept
+                                                                           ptr_t const ptr) noexcept
 {
     if (static_cast<segment_id_underlying_t>(id) == NULL_POINTER_ID)
     {
         return NULL_POINTER_OFFSET;
     }
-    auto* basePtr = getBasePtr(id);
-    // NOLINTJUSTIFICATION Cast needed for pointer arithmetic
+    const auto* const basePtr = getBasePtr(id);
+    // AXIVION Next Construct AutosarC++19_03-A5.2.4, AutosarC++19_03-M5.2.9 : Cast needed for pointer arithmetic
     // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
     return reinterpret_cast<offset_t>(ptr) - reinterpret_cast<offset_t>(basePtr);
```

Comment:
This is weird. Why does this cast even work, as it tries to cast `const` away (should not work with `reinterpret_cast`). Probably because it is a cast to a non-ptr type and we do not try (cannot) access the data from the destination type.

It is correct, but the const required by Axivion is really too much, semi-defeats the purpose of `auto` (the brevity).

Nothing to do if we want the rule, but enforcing `const` for local variables (that are not changed) is not a good idea IMO. They should be optional for readability of longer functions.